connector/src/main/scala/com/microsoft/kusto/spark/sql/extension/SparkExtension.scala

```diff
@@ -38,7 +36,24 @@ object SparkExtension {
         .option(KustoSinkOptions.KUSTO_DATABASE, database)
         .option(KustoSinkOptions.KUSTO_TABLE, table)
         .options(conf)
+        .mode(SaveMode.Append)
         .save()
     }
   }
-}
\ No newline at end of file
+
+  implicit class DataStreamWriterExtension(df: DataStreamWriter[Row]) {
+    def kusto(kustoCluster: String, database: String, table: String, conf: Map[String, String] = Map.empty[String, String], sparkIngestionProperties: Option[SparkIngestionProperties] = None): Unit = {
+
+      (if (sparkIngestionProperties.isDefined) {
```

Comment:
not exactly sure I understand this.. if you get properties as json you use them, otherwise you use params? shouldn't params be optional as well if so?

Reply:
sparkIngestionProperties is only one option that can pass in the conf options map.
its not the same thing .
you missed the logic here - the line starting with "}).format" is executed anyway , maybe i should break it into two if its misleading

Reply:
Oh I see, although I don't know why not just

```scala
if (sparkIngestionProperties.isDefined) {
  df.option(KustoSinkOptions.KUSTO_SPARK_INGESTION_PROPERTIES_JSON, sparkIngestionProperties.get.toString)
}

df.format("com.microsoft.kusto.spark.datasource")
        .option(KustoSinkOptions.KUSTO_CLUSTER, kustoCluster)
        .option(KustoSinkOptions.KUSTO_DATABASE, database)
        .option(KustoSinkOptions.KUSTO_TABLE, table)
        .options(conf)